Lib/asyncio/runners.py

```diff
@@ -118,7 +120,14 @@ def run(self, coro, *, context=None):
                 events.set_event_loop(self._loop)
             return self._loop.run_until_complete(task)
         except exceptions.CancelledError:
-            if self._interrupt_count > 0 and task.uncancel() == 0:
+            uncancel = getattr(task, "uncancel", None)
```

Comment:
If this is called from asyncio.Runner rather than asyncio.run then we know we can rely on uncancel

Reply:
The additional complexity is not worth it, better to just warn on both `asyncio.run`  and `asyncio.Runner` when `uncancel` is missing.

Reply:
I'm not suggesting any additional complexity, I'm suggesting decoupling asyncio.run and asyncio.Runner and having the new behavior only in Runner

Reply:
How would you do that? by adding a flag or duplicating code for `asyncio.run` and `asyncio.Runner`?

Reply:
yeah, asyncio.run would use the 3.10 code and asyncio.Runner would use the new 3.11 code

Reply:
> yeah, asyncio.run would use the 3.10 code and asyncio.Runner would use the new 3.11 code

Exactly, this is the complexity I want to avoid.